File: DEC 2025-JAN 2026/1-TwoSum/1-TwoSum.js
// Last updated: 16/02/2026, 16:15:24
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let maps=new Map();
    for(let i=0;i<nums.length;i++){
        let complement=target-nums[i];
        if(maps.has(complement)){
            return [maps.get(complement),i];
        }
        maps.set(nums[i],i)
    }
};

File: 1-TwoSum/1-TwoSum.py
# Last updated: 16/02/2026, 16:15:25
class Solution(object):
    def twoSum(self, nums, target):
    
        d={}
        for i in range(0,len(nums)):
            value=nums[i]
            diff=target-value
            if value not in d:
                d[diff]=i 
            else:
                ci=i
                pi=d[value]
                return [ci,pi]


        